support multiple namespaces - breaking change

balonik opened the pull request:
This is a BREAKING CHANGE, because the type of `namespaceSelector` is changed from `string` to `array`

Rewrite the `MutatingWebhookConfiguration` to allow adding multiple namespaces.
Fixes https://github.com/danfromtitan/envars-from-node-labels/issues/16

balonik committed:
support multiple namespaces

Signed-off-by: Peter Jakubis <balonik32@gmail.com>

danfromtitan commented:
Do you mind increasing the helm chart version to reflect the breaking change (i.e. make it 0.2.0)

danfromtitan commented on charts/envars-webhook/templates/mutatingwebhook.yaml:
```diff
@@ -8,7 +8,7 @@ metadata:
     {{- include "envars-webhook.labels" . | nindent 4 }}
   annotations:
-    "helm.sh/hook": "pre-install"
+    "helm.sh/hook": "post-install,post-upgrade,pre-delete"
```
Could you please explain the reasoning behind changing the order of resource creation. I vaguely remember the webhook had to be created before anything else.

balonik replied:
I changed it, because the Webhook is using the `Service` which is using the `Deployment` and there is no point in having Webhook before any `Pod` is actually running. It will just cause any Webhook calls by k8s control-plane to fail, because there is nothing that could respond. I can't think of a reason why would a mutatingwebhook needed to exist first.

balonik commented:
Actually, almost immediately after I implemented this change it got to me that I could just easily put a label with same value on all namespaces and the result will be the same. This change won't be needed really. Optionally, I would just change the `values.yaml` a bit to make the names more straight forward and adjusted the `mutatingwebhookconfiguration` accordingly. I'd change the `namespaceSelector` and the recently added `namespaceSelectorLabel` into:

```
namespaceSelector:
  labelKey: name
  labelValue: sample
```

This change would make sense if someone needs to use existing label that has different value on namespaces, though. What do you think?

danfromtitan commented:
> This change would make sense if someone needs to use existing label that has different value on namespaces, though. What do you think?

That would obscure the configuration, I like using the list of namespaces because it's more explicit.

Please increase the chart version and I'll merge.

balonik commented:
Done.

danfromtitan commented:
Tests passed, thank you for your contribution.
